File: mobile/src/main/java/net/mm2d/android/util/RuntimeEnvironment.java
package net.mm2d.android.util;

import android.os.Build;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class RuntimeEnvironment {

    /**
     * 現在の実行状態がエミュレータか否かを判定する。
     *
     * @return true:エミュレータである。false:それ以外
     */
    public static boolean isEmulator() {
        return Build.UNKNOWN.equals(Build.BOOTLOADER)
                && Build.MODEL.contains("Android SDK");
    }
}

File: mobile/src/main/java/net/mm2d/android/util/RuntimeEnvironment.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.util

import android.os.Build

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
object RuntimeEnvironment {
    /**
     * 現在の実行状態がエミュレータか否かを判定する。
     *
     * @return true:エミュレータである。false:それ以外
     */
    val isEmulator: Boolean by lazy {
        Build.UNKNOWN == Build.BOOTLOADER && Build.MODEL.contains("Android SDK")
    }
}
